streams/cluster/src/test/java/com/hortonworks/streamline/streams/cluster/register/impl/StormServiceRegistrarTest.java

```diff
@@ -57,38 +61,62 @@ public void testRegister() throws Exception {
 
         StormServiceRegistrar registrar = initializeServiceRegistrar();
 
-        try (InputStream is = getClass().getClassLoader().getResourceAsStream(STORM_YAML_FILE_PATH)) {
-            Config config = new Config();
-            config.put(StormServiceRegistrar.PARAM_STORM_UI_SERVER_HOSTNAME, "storm-1");
-            config.put(StormServiceRegistrar.PARAM_NIMBUS_HOSTNAMES, Lists.newArrayList("storm-1", "storm-2"));
-            config.put(StormServiceRegistrar.PARAM_NIMBUS_PRINCIPAL_NAME, "nimbus/_HOST@EXAMPLE.COM");
-            ManualServiceRegistrar.ConfigFileInfo stormYaml = new ManualServiceRegistrar.ConfigFileInfo(STORM_YAML, is);
-            registrar.register(cluster, config, Lists.newArrayList(stormYaml));
-        }
+        // explicit convert Object
+        Config config = new Config();
+        config.put(StormServiceRegistrar.PARAM_NIMBUS_SEEDS, "storm-1,storm-2");
+        config.put(StormServiceRegistrar.PARAM_NIMBUS_THRIFT_PORT, (Object) 6627);
+        config.put(StormServiceRegistrar.PARAM_UI_HOST, "storm-1");
+        config.put(StormServiceRegistrar.PARAM_UI_PORT, (Object) 8080);
+        config.put(StormServiceRegistrar.PARAM_NIMBUS_THRIFT_MAX_BUFFER_SIZE, (Object) 102476800);
+        config.put(StormServiceRegistrar.PARAM_THRIFT_TRANSPORT, "org.apache.storm.security.auth.SimpleTransportPlugin");
+        config.put(StormServiceRegistrar.PARAM_PRINCIPAL_TO_LOCAL, "org.apache.storm.security.auth.DefaultPrincipalToLocal");
+        config.put(StormServiceRegistrar.PARAM_NIMBUS_PRINCIPAL_NAME, "nimbus/_HOST@EXAMPLE.COM");
+        registrar.register(cluster, config, Collections.emptyList());
 
         Service stormService = environmentService.getServiceByName(cluster.getId(), Constants.Storm.SERVICE_NAME);
         assertNotNull(stormService);
+
+        Component nimbus = environmentService.getComponentByName(stormService.getId(), ComponentPropertyPattern.NIMBUS.name());
+        assertNotNull(nimbus);
+
+        Collection<ComponentProcess> nimbusProcesses = environmentService.listComponentProcessesInComponent(nimbus.getId());
+        List<String> hosts = nimbusProcesses.stream().map(ComponentProcess::getHost).collect(Collectors.toList());
+        assertEquals(Sets.newHashSet("storm-1", "storm-2"), new HashSet<>(hosts));
+        List<Integer> ports = nimbusProcesses.stream().map(ComponentProcess::getPort).collect(Collectors.toList());
+        assertEquals(Sets.newHashSet(6627, 6627), new HashSet<>(ports));
```

Comment:
Is this always true, or does it depend on the order of the sets? If so it could cause sporadic test failures.